### ISSUE TITLE
Allow `require` in object expressions under the `global-require` rule

### ISSUE THREAD

**Issue by maxtruxa** [eslint/eslint#5773]
**What version of ESLint are you using?**
2.5.3

**What rule do you want to change?**
`global-require`

**What code should be flagged as correct with this change?**
```js
/*eslint global-require: "error"*/
var foo = {bar: require('foo.bar')};
```

**What happens when the rule is applied to this code now?**
```sh
error  Unexpected require()  global-require
```

**Why would you want to write code like that?**
Some libraries (like `lodash`) offer standalone modules for single functions (like `lodash.assign`).
Using this syntax you can then use `_.assign` as if you had the full library available.

Note that the example below is currently not being flagged (while doing essentially the same thing):
```js
/*eslint global-require: "error"*/
var foo = {};
foo.bar = require('foo.bar');
```

**Comment by mysticatea** [comment]
Thank you for this issue.

I confirmed this on the online demo.
Looks like a bug since object literals don't make any scope.

**Comment by maxtruxa** [comment]
If you want me to, I can create a pull request later today. The fix is pretty straight forward.

**Comment by mysticatea** [comment]
@maxtruxa Thank you so much. Of course, we welcome the patch!

**Comment by nzakas** [comment]
I mentioned on the PR, but this is *not* a bug. This rule is meant to emulate where `import` can be used, not necessarily about the global scope. We also don't allow require inside of if statements, so this change doesn't make sense.

**Comment by ilyavolodin** [comment]
@nzakas `import` can't be used inside object expression?

**Comment by nzakas** [comment]
Nope

**Comment by michaelficarra** [comment]
So can we close this?

**Comment by ilyavolodin** [comment]
This looks like it's behaving as expected. Closing, since this rule is created to emulate `import`.

**Comment by maxtruxa** [comment]
Agreed. I'll actually go with the alternative syntax that isn't flagged anyways, because after thinking about it a bit longer I decided it looks cleaner.

```
var foo = {};
foo.bar = require('foo.bar');
```

**Comment by jacksonrayhamilton** [comment]
Just encountered this myself. I think the motivation for this rule should be re-evaluated. I would like to use this rule for the following justifications, as stated in its documentation:

- Make it easier to identify dependencies.
- Avoid synchronous calls.

However, I see conflicts in this justification:

- Emulate `import`.

If I were a Browserify or WebPack user, and relied on static analysis to resolve my imports, or if I planned to eventually migrate my Node modules to ES6 modules, then this synonymous treatment might be useful.

However, `require()` is not `import`. In Node.js, we actually do have the freedom to call `require()` in an `if` block or an object expression, which can be useful. If I am just requiring Node.js modules in the traditional manner, why should `require()` be restricted in the same way as `import`?

I think the extra restriction on "top-level statements" should at least be configurable as an option. If it were an option, I think it should also be turned off by default, since an "authentic" Node.js / CommonJS environment actually doesn't have this "top-level statement" restriction.

**Comment by nzakas** [comment]
I'm sorry, we are not accepting changes for this rule. If you want, you can create a custom rule that does exactly what you want.
